ext/adaptors/ranges-lib.tex

```diff
@@ -544,40 +559,43 @@
 
 \indexlibrary{\idxcode{subrange}!\idxcode{subrange}}%
 \begin{itemdecl}
-template <ConvertibleTo<I> X, ConvertibleTo<S> Y, subrange_kind B>
-constexpr subrange(subrange<X, Y, B> r, difference_type_t<I> n) requires bool(Sized);
+template <ConvertibleTo<I> X, ConvertibleTo<S> Y, subrange_kind Z>
+constexpr subrange(subrange<X, Y, Z> r, difference_type_t<I> n)
+  requires K == subrange_kind::sized;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\requires \tcode{bool(B)} is \tcode{false} or \tcode{n == r.size()}.
-
-\pnum
```

Comment:
Because those requirements are already necessary for `subrange{r.begin(), r.end(), n}`?

Reply:
Yes, "Effects equivalent to" inherits requirements ([[structure.specifications]/4](http://eel.is/c++draft/structure.specifications#4)).